openlibrary/core/models.py

```diff
@@ -211,12 +211,12 @@ def canonicalize(edition):
         work = edition.works and edition.works[0]
 
         def _get_cover_url(edition, work):
-            ol_covers = (
+            ol_covers = [
                 doc.get_cover().url('M') for doc in [edition, work]
                 if doc and doc.get_cover()
-            )
+            ]
             if ol_covers:
-                return next(ol_covers)
+                return ol_covers[0]
```

Comment:
The current implementation only processes the loop until the first element is created.  The proposed change creates the entire list and then only uses the first element.  Is this extra processing desired and useful?